essentials/src/telemetry_subscription.rs

```diff
@@ -84,27 +133,84 @@ impl TelemetrySubscription {
 	pub async fn run(
 		self,
 		url: String,
-		chain_hash: H256,
 		shutdown_tx: BroadcastSender<()>,
 	) -> color_eyre::Result<Vec<tokio::task::JoinHandle<()>>> {
 		Ok(self
 			.consumers
 			.into_iter()
 			.map(|update_channel| {
-				tokio::spawn(Self::run_per_consumer(update_channel, url.clone(), chain_hash, shutdown_tx.clone()))
+				tokio::spawn(Self::run_per_consumer(update_channel, url.clone(), shutdown_tx.clone()))
 			})
 			.collect::<Vec<_>>())
 	}
 }
 
-async fn telemetry_stream(url: &str, chain_hash: H256) -> WebSocketStream<MaybeTlsStream<TcpStream>> {
-	let url = Url::parse(url).unwrap();
-	let (mut ws_stream, _) = connect_async(url).await.expect("Failed to connect");
-	if let Err(e) = ws_stream.send(Message::Text(format!("subscribe:{:?}", chain_hash))).await {
-		info!("Cannot subscribe to chain with hash {}: {:?}", chain_hash, e);
+const CHAINS_CHUNK_SIZE: usize = 10;
+
+async fn choose_chain(chains: &HashMap<H256, AddedChain>) -> color_eyre::Result<H256, ()> {
+	let list: Vec<AddedChain> = chains
+		.iter()
+		.map(|(_, v)| v.clone())
+		.sorted_by_key(|c| Reverse(c.node_count))
+		.collect();
+	let list_size = list.len();
+
+	match list_size {
+		0 => {
+			println!("No chains found...");
+			return Err(())
+		},
+		1 => {
+			println!("Found 1 chain.\n{}", list[0]);
+			return Ok(list[0].genesis_hash)
+		},
+		size => {
+			println!("Found {} chains.\n", size);
+		},
```

Comment:
See comment above for the `Display` trait. Error should also be meaningful and not a tuple, as in general, it is a caller responsibility what to do with a `Result` returned.